### PR TITLE
bazel: unit test and upload build results

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

**Comment by spxtr** [comment]
This looks reasonable to me.

File: jobs/pull-kubernetes-bazel.sh [hunk 1]
```diff
@@ -28,14 +28,16 @@ release-1.0|release-1.1|release-1.2|release-1.3|release-1.4|release-1.5)
   ;;
 esac
 
-# TODO(ixdy): remove once all branches are back to build/
-build_dir=build
-if [[ -d build-tools/ ]]; then
-  build_dir=build-tools
+bazel build //cmd/... //pkg/... //federation/... //plugin/... //build/... //test/... && rc=$? || rc=$?
```
Comment (mikedanese) [inline jobs/pull-kubernetes-bazel.sh]: fixed

**Comment by mikedanese** [comment]
Added some

File: jobs/pull-kubernetes-bazel.sh [hunk 2]
```diff
@@ -28,14 +28,16 @@ release-1.0|release-1.1|release-1.2|release-1.3|release-1.4|release-1.5)
   ;;
 esac
 
-# TODO(ixdy): remove once all branches are back to build/
-build_dir=build
-if [[ -d build-tools/ ]]; then
-  build_dir=build-tools
+bazel build //cmd/... //pkg/... //federation/... //plugin/... //build/... //examples/... //test/... //third_party/... && rc=$? || rc=$?
+
+if [[ "${rc}" != 0 ]]; then
```
Comment (ixdy) [inline jobs/pull-kubernetes-bazel.sh]: shouldn't these be `${rc} == 0`?

Comment (spxtr) [inline jobs/pull-kubernetes-bazel.sh]: Yes, they should.

Comment (mikedanese) [inline jobs/pull-kubernetes-bazel.sh]: This produces the intended behavior and aligns with the style guide which days to quote all variable expansions unless it's necessary not to

Comment (ixdy) [inline jobs/pull-kubernetes-bazel.sh]: that's not what I meant. I meant these should be ==, not !=.

Comment (mikedanese) [inline jobs/pull-kubernetes-bazel.sh]: Lol Gah, thanks

**Comment by spxtr** [comment]
/lgtm

Assuming you're confident that these pass.

File: jobs/pull-kubernetes-bazel.sh [hunk 3]
```diff
@@ -28,14 +28,16 @@ release-1.0|release-1.1|release-1.2|release-1.3|release-1.4|release-1.5)
...
 esac
 
-# TODO(ixdy): remove once all branches are back to build/
-build_dir=build
-if [[ -d build-tools/ ]]; then
-  build_dir=build-tools
+bazel build //cmd/... //pkg/... //federation/... //plugin/... //build/... //examples/... //test/... //third_party/... && rc=$? || rc=$?
+
+if [[ "${rc}" == 0 ]]; then
+  bazel test --test_tag_filters '-skip' //cmd/... //pkg/... //federation/... //plugin/... //build/... //third_party/... && rc=$? || rc=$?
```
Comment (ixdy) [inline jobs/pull-kubernetes-bazel.sh]: do we get junit from tests saved anywhere?

Comment (spxtr) [inline jobs/pull-kubernetes-bazel.sh]: They're saved under `bazel-testlogs`, which I copy into `_artifacts` later in this script.

**Pull request merged**